tv-app/effecticeTV/app/src/main/java/band/effective/office/tv/repository/impl/SynologyPhotoImpl.kt

```diff
@@ -1,92 +1,77 @@
+import band.effective.office.tv.core.network.entity.Either
+import band.effective.office.tv.core.network.entity.ErrorReason
+import band.effective.office.tv.core.network.entity.unpack
 import band.effective.office.tv.network.synology.SynologyApi
-import band.effective.office.tv.network.synology.response.SynologyAuthResponse
-import band.effective.office.tv.network.synology.response.SynologyListResponse
+import band.effective.office.tv.network.synology.models.response.SynologyAuthResponse
+import band.effective.office.tv.network.synology.models.response.SynologyListResponse
 import band.effective.office.tv.repository.SynologyPhoto
-import com.squareup.moshi.Moshi
-import okhttp3.Response
-import okhttp3.ResponseBody
+import java.net.URLEncoder
 import javax.inject.Inject
 
 class SynologyPhotoImpl @Inject constructor(
     private val synologyApi: SynologyApi,
-) : SynologyPhoto, BaseRepository() {
+) : SynologyPhoto {
     private var dirPath: String = ""
     private var isAuth: Boolean = false
-    override suspend fun getPhotos(): Resource<List<Photo>> {
-        TODO("Not yet implemented")
-    }
-
-    override suspend fun changeDir(dir: String): Resource<Boolean> {
-        TODO("Not yet implemented")
-    }
+    private var sid: String = ""
 
-    override suspend fun getPhotoPath(folderPath: String): Resource<List<String>> {
+    override suspend fun getPhotosUrl(folderPath: String): Resource<List<String>> {
         var authResponse: Resource<Boolean>? = null
         if (!isAuth) authResponse = auth()
 
-        if (authResponse is Resource.Data) {
-            val res: Resource<SynologyListResponse> = safeApiCall {
+        if (isAuth) {
+            val res: Either<ErrorReason, SynologyListResponse> =
                 synologyApi.getFiles(
-                    version = 3,
+                    sid = sid,
+                    version = 1,
                     method = "list",
                     folderPath = folderPath
                 )
-            }
-            return when (res) {
-                is Resource.Data -> Resource.Data(res.data.data.files.map { it.path })
-                is Resource.Error -> Resource.Error(res.error)
-                else -> Resource.Loading()
-            }
-        }
-
-        return if (authResponse is Resource.Error) {
-            val message = authResponse.error
-            Resource.Error(message)
-        } else Resource.Loading()
 
+            return res.unpack(
+                success = { response ->
+                    val paths = response.data.files.map { formThumbPhoto(it.path) }
+                    Resource.Data(paths)
+                },
+                error = { error ->
+                    Resource.Error(error.message ?: "")
+                }
+            )
+        }
+        val message = (authResponse as Resource.Error).error
+        return Resource.Error(message)
     }
 
     override suspend fun auth(): Resource<Boolean> {
```

Comment:
- Separate authorization and synology photo by two different repositories
- Create UseCase for authorization and Synology photo request